apps/test_libc.c

```
@@ -0,0 +1,101 @@
+        printf("  p1 content: \"%s\"\n", (char *)p1);
+    }
+
+    free(p1);
+    free(p2);
+    free(p3);
+    printf("  free(p1,p2,p3) done\n");
+
+    /* Realloc */
+    void *p4 = malloc(32);
+    if (p4) memset(p4, 'B', 31);
+    p4 = realloc(p4, 64);
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)